module/VuFind/src/VuFind/Auth/AbstractBase.php

```diff
@@ -572,4 +573,43 @@ protected function setUserValueByField(UserEntityInterface $user, string $field,
         }
         $user->$setter($value);
     }
+
+    /**
+     * Save user and any ILS credentials.
+     *
+     * Also updates user card data library cards are if enabled.
+     *
+     * @param UserEntityInterface $user             User
+     * @param ?string             $catPassword      ILS catalog password
+     * @param ILSAuthenticator    $ilsAuthenticator ILS authenticator
+     *
+     * @return void
+     */
+    protected function saveUserAndCredentials(
+        UserEntityInterface $user,
+        ?string $catPassword,
+        ILSAuthenticator $ilsAuthenticator
```

Comment:
I decided to pass ILSAuthenticator as a parameter so that no constructor changes were needed and this method was kept really standalone.